src/components/Modal.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import { createPortal } from "react-dom";
+import { useRecoilValue } from "recoil";
+import { themeState } from "../store";
+import { motion } from "framer-motion";
+import Button, { ButtonType } from "./Button";
+import { faCancel } from "@fortawesome/free-solid-svg-icons";
+
+type ModalProps = React.PropsWithChildren & {
+  title: string;
+  onClose: () => void;
+  onClickPrimary?: () => void;
+  onClickSecondary?: () => void;
+  primaryClickDisabled?: boolean;
+  secondaryClickDisabled?: boolean;
+  primaryClickText?: string;
+  secondaryClickText?: string;
+  containerClassName?: string;
+  headerClassName?: string;
+  height?: string;
+  width?: string;
+};
+
+const Modal: React.FC<ModalProps> = ({
+  children,
+  title,
+  onClose,
+  onClickPrimary,
+  primaryClickDisabled,
+  primaryClickText,
+  onClickSecondary,
+  secondaryClickDisabled,
+  secondaryClickText,
+  containerClassName,
+  headerClassName,
+  height,
+  width,
+}) => {
+  const theme = useRecoilValue(themeState);
+  const defaultFontColor = theme ? "text-white" : "text-black";
+  const defaultModalHeaderBackgroundColor = theme ? "bg-zinc-700" : "bg-zinc-200";
+  const defaultBodyBackgroundColor = theme ? "bg-zinc-800" : "bg-white";
+  const defaultPageTheme = theme ? "dark" : "light";
+
+  const modalPageClassName = `fixed flex items-center justify-center inset-0 overflow-hidden ${defaultPageTheme} ${
+    theme ? "bg-zinc-800/70" : "bg-zinc-400/60"
+  }`;
+
+  const modalContainerClassName = `fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 flex flex-col shadow-lg`;
+
+  const modalHeaderClassName = `text-lg font-semibold rounded-t-lg p-2
+  ${defaultModalHeaderBackgroundColor} ${defaultFontColor} ${headerClassName}`;
+
+  const modalBodyClassName = `p-4 ${defaultFontColor} ${height || "h-[30em]"} 
+  ${width || "w-[25em]"} ${defaultBodyBackgroundColor} ${containerClassName}`;
+
+  const modalFooterClassName = `rounded-b-lg p-2 min-h-[2.5em] flex gap-1 justify-end ${defaultModalHeaderBackgroundColor} ${defaultFontColor}`;
```

Comment:
not a huge fan of defining styles as variables when it isn't being reused but I suppose it is preference >.<

Reply:
Two reasons they're defined here: 1, defining them inside of the html is terrible ugly. 2, they *SHOULD* be memoized based on theme, but I did not do that (how often is theme being changed anyways XD)